### PR TITLE
Updated course model and commited related changes

### PR DESCRIPTION
**Pull request by DanWaldman**
in this PR we:

- Figured out we were missing a foreign key of university in course model - which is crucial for getting the courses of a certain degree of a certain university - so we added the foreign key in the model.
- Added function "course_belongs" to course model to check if a course belongs to a certain degree in a certain university.
- Changed migration files to fit the new foreign key in course model.
- Added test for the new function "course_belongs".
- Removed all fixtures from test files that were in conftest.

#114

### CODE REVIEW

File: AcadeMeData/models.py [hunk 1]
```diff
@@ -114,6 +114,9 @@ def get_university_by_location(location):
         # gets the relevant university that match the given location
         return University.objects.get(location=location)
 
+    def return_univerities(self):
+        return
```
Comment (razashtar93) [inline AcadeMeData/models.py:118]: empty return?

**Review by razashtar93 — APPROVED** [review]
LGTM

File: AcadeMeData/models.py (hunk 1 above)
Comment (DanWaldman) [reply AcadeMeData/models.py]: fixed, my bad

File: AcadeMeData/tests/test_course.py [hunk 2]
```diff
@@ -17,6 +15,13 @@ def generate_course(generate_degree, generate_professor, course_id=1, name="Hist
 class TestCourseModel:
     def test_get_course_by_name(self, generate_course, name="History of Countries"):
         course = Course.get_course_by_name(name)
-
         assert course.name == generate_course.name
         assert course.professor == generate_course.professor
+
+    def test_if_course_belongs(self, generate_course, generate_degree, generate_university):
+        bool = generate_course.course_belongs(generate_university, generate_degree)
```
[the remark this answers is not in the document]
Comment (DanWaldman) [reply AcadeMeData/tests/test_course.py:22]: assert True==generate_course.course_belongs(
generate_university,
generate_degree)

is the same as:

assert generate_course.course_belongs(
generate_university,
generate_degree)

but I will get rid of the bool and do it directly.

[the remark this answers is not in the document]
Comment (DanWaldman) [reply AcadeMeData/tests/test_course.py]: Done

File: AcadeMeData/tests/test_course.py [hunk 3]
```diff
@@ -17,6 +15,13 @@ def generate_course(generate_degree, generate_professor, course_id=1, name="Hist
...
+        assert bool
+
+    def test_if_course_not_belongs(self, generate_course, generate_university):
+        bool = generate_course.course_belongs(generate_university, None)
+        assert not bool
```
[the remark this answers is not in the document]
Comment (DanWaldman) [reply AcadeMeData/tests/test_course.py]: Done

**Review by DanielItkes — APPROVED** [review]

**Review by Yarboa — APPROVED** [review]

**Review by DeanCivlin — APPROVED** [review]

File: AcadeMeData/models.py [hunk 4]
```diff
@@ -239,3 +241,11 @@ def get_course_by_name(name):
             return course
         except Course.DoesNotExist:
             return None
+
+    def course_belongs(self, university, degree):
+        """
+        returns if course belongs to this degree in this university
+        """
+        if degree in self.degree.all() and university == self.university:
```
Comment (Omeramsc) [inline AcadeMeData/models.py:249]: Using [Truthy and Falsy](https://www.geeksforgeeks.org/truthy-vs-falsy-values-in-python/) values is considered more pythonic than returning True\False like this.

try to modify the function so it'll turn into a one-liner :)

File: AcadeMeData/tests/test_course.py [hunk 5]
```diff
@@ -17,6 +15,11 @@ def generate_course(generate_degree, generate_professor, course_id=1, name="Hist
...
         course = Course.get_course_by_name(name)
-
         assert course.name == generate_course.name
         assert course.professor == generate_course.professor
+
+    def test_if_course_belongs(self, generate_course, generate_degree, generate_university):
+        assert generate_course.course_belongs(generate_university, generate_degree)
+
+    def test_if_course_not_belongs(self, generate_course, generate_university):
+        assert not generate_course.course_belongs(generate_university, None)
```
Comment (Omeramsc) [inline AcadeMeData/tests/test_course.py:25]: The test is insufficient, you should check that with a degree that does not include the course, not with None.

**Review by Omeramsc — CHANGES_REQUESTED** [review]

File: AcadeMeData/models.py (hunk 4 above)
Comment (DanWaldman) [reply AcadeMeData/models.py]: Turned it to a one-liner @Omeramsc

File: AcadeMeData/tests/test_course.py (hunk 5 above)
Comment (DanWaldman) [reply AcadeMeData/tests/test_course.py]: Added a degree instead of none @Omeramsc

**Comment by DanWaldman** [comment]
@Omeramsc

File: AcadeMeData/tests/test_course.py [hunk 6]
```diff
@@ -1,22 +1,34 @@
...
-    course = Course.create_course(course_id=course_id, name=name, degree=degree, mandatory=mandatory,
-                                  description=description, professor=professor)
+    course = Course.create_course(course_id=course_id, name=name, degree=generate_degree, mandatory=mandatory,
+                                  description=description, professor=generate_professor, university=generate_university)
     course.save()
     return course
 
 
+@pytest.fixture
+def generate_degree2(degree_id=5, name='Computer Science', universities="Reichman University",
```
Comment (Omeramsc) [inline AcadeMeData/tests/test_course.py:15]: Please give it a more informative name. what differs between this and `generate_course`? (I know the answer, write it so other readers will know too)

**Review by Omeramsc — COMMENTED** [review]
overall LGTM, please address my last comment.

**Review by Yarboa — APPROVED** [review]
LGTM, please note @Omeramsc comment

File: AcadeMeData/tests/test_course.py (hunk 6 above)
Comment (DanWaldman) [reply AcadeMeData/tests/test_course.py]: Done @Omeramsc

**Comment by DanWaldman** [comment]
> LGTM, please note @Omeramsc comment

Resolved

**Review by Omeramsc — APPROVED** [review]
LGTM

**Pull request merged**